src/sql/src/plan/lowering.rs

```diff
@@ -1686,7 +1699,10 @@ impl HirScalarExpr {
                 els: Box::new(els.lower_uncorrelated()?),
             },
             Select { .. } | Exists { .. } | Parameter(..) | Column(..) | Windowing(..) => {
-                sql_bail!("unexpected ScalarExpr in uncorrelated plan: {:?}", self);
+                sql_bail!(
+                    "Internal error: unexpected HirScalarExpr in lower_uncorrelated: {:?}",
+                    self
+                );
```

Comment:
This was already kind of an internal error in the sense that it was totally not understandable for users. I've now made this official, by explicitly saying that this is an "Internal error".

There are many calls to this function, and I'm planning to go through all of them, to sanity check them that they avoid this error.